src/data/data_fetcher.py

```diff
@@ -273,3 +276,30 @@ def get_transactions_on_page(self):
         self.last_contribution_receipt_date = self.info["pagination"]["last_indexes"][
             "last_contribution_receipt_date"
         ]
+
+    def _build_df(self):
+        self.df = pd.DataFrame(
+            self.complete_list,
+            columns=[
+                "contributor_occupation",
+                "contributor_employer",
+                "contributor_city",
+                "contributor_state",
+                "contributor_zip",
+                "party",
+            ],
+        )
+        self.df.fillna(value="", inplace=True)
+        return self.df
+
+    def _save_df_data(self):
```

Comment:
I'd un-`_` this one. Saving the data is useful as heck in the top-level function!